pkg/sql/information_schema.go

```diff
@@ -1165,17 +1165,19 @@ CREATE TABLE information_schema.schema_privileges (
 	populate: func(ctx context.Context, p *planner, dbContext catalog.DatabaseDescriptor, addRow func(...tree.Datum) error) error {
 		return forEachDatabaseDesc(ctx, p, dbContext, true, /* requiresPrivileges */
 			func(db catalog.DatabaseDescriptor) error {
-				return forEachSchema(ctx, p, db, func(sc catalog.ResolvedSchema) error {
+				return forEachSchema(ctx, p, db, func(sc catalog.SchemaDescriptor) error {
 					var privs []descpb.UserPrivilegeString
-					if sc.Kind == catalog.SchemaUserDefined {
+					if sc.SchemaKind() == catalog.SchemaUserDefined {
 						// User defined schemas have their own privileges.
-						privs = sc.Desc.GetPrivileges().Show(privilege.Schema)
+						privs = sc.GetPrivileges().Show(privilege.Schema)
 					} else {
 						// Other schemas inherit from the parent database.
+						// TODO(ajwerner): Fix this because it's bogus for everything other
+						// than public.
```

Comment:
Implementing `GetPrivileges` for synthetic schemas would obviate the need for branching on `SchemaKind()` altogether, it seems.